src/Controller/Component/CommonComponent.php

```diff
@@ -25,8 +26,15 @@ class CommonComponent extends Component {
 	 */
 	public function startup(Event $event) {
 		// Data preparation
-		if ($this->Controller->request->getData() && !Configure::read('DataPreparation.notrim')) {
-			$this->Controller->request->data = Utility::trimDeep($this->Controller->request->getData());
+		if ($this->Controller->getRequest()->getData() && !Configure::read('DataPreparation.notrim')) {
+			$request = $this->Controller->getRequest();
+			$new_data = Hash::flatten(Utility::trimDeep($this->Controller->getRequest()->getData()));
```

Comment:
newData

Reply:
whats the hash flatten here?

Reply:
- [x] newData
- [x] HashFlatten: Was over thinking things its been removed